features/environment.py

```diff
@@ -202,38 +207,70 @@ def create_uat_lxd_image(context: Context, series: str) -> None:
         )
         return
     now = datetime.datetime.now()
+    ubuntu_series = "ubuntu-daily:%s" % series
+
+    #if we are building the package from PR, we need a 2nd_base_image
+    #without the dirty build dependencies from that 1st_base_image
+
+    if context.config.build_pr:
+        #creating a new image name for debugging purpose
+        build_container_name = "behave-image-pre-build-%s-" % series + now.strftime("%s%f")
+        launch_lxd_container(context, ubuntu_series, build_container_name)
+        lxc_push_source_pull_deb_pkg(build_container_name)
+        #here it should have the new built .deb @/tmp dir
+        #/tmp/ubuntu-advantage-tools_20.4_amd64.deb
+
+    build_container_name = "behave-image-build-%s-" % series + now.strftime(
+            "%s%f"
+        )
+
+    launch_lxd_container(context, ubuntu_series, build_container_name)
+
+    #if build_pr it will install new built .deb
+    _install_uat_in_container(build_container_name, context.config.build_pr)
+
     context.series_image_name[
```

Comment:
Looking over this in more depth, it seems we actually don't have to create a separate build vm anyway. I hadn't thought enough about it, but we can re-start the build VM after we have already taken a snapshot then dirty the buildVM because we are no longer using it.

So I think this section could instead look like:
```
    context.series_image_name[                                                  
        series                                                                  
    ] = "behave-image-%s-" % series + now.strftime("%s%f")                      
    build_container_name = "behave-image-build-%s-" % series + now.strftime(    
        "%s%f"                                                                  
    )                                                                           
    ubuntu_series = "ubuntu-daily:%s" % series                                  
    launch_lxd_container(context, ubuntu_series, build_container_name)          
    _install_uat_in_container(build_container_name)                             
    _capture_container_as_image(                                                
        build_container_name, context.series_image_name[series]                 
    )                                                                           
    if context.config.build_pr:                                                 
        subprocess.run(["lxc", "start", build_container_name])                  
        lxc_build_deb(                                                          
            build_container_name, output_deb_path="/tmp/ubuntu-advantage.deb")  
        subprocess.run(["lxc", "stop", build_container_name])                  
```

Reply:
I kept both images because this way, the snapshot doesn't have the proper UA-tools.deb (from the source) installed in. So, let me know what do you think about the new modifications I've made.
If you'd like to keep as you suggested, we need the install the .deb for each test that we run the scenarios - when creating thebehave-test-*...